v03_pipeline/lib/methods/relatedness.py

```diff
@@ -1,55 +1,38 @@
 from __future__ import annotations
 
-from typing import TYPE_CHECKING
-
 import hail as hl
 from gnomad.sample_qc.pipeline import filter_rows_for_qc
 
-from v03_pipeline.lib.model import CachedReferenceDatasetQuery
-from v03_pipeline.lib.paths import valid_cached_reference_dataset_query_path
-
-if TYPE_CHECKING:
-    from v03_pipeline.lib.model import ReferenceGenome
-
 
 def filter_and_ld_prune(
     mt: hl.MatrixTable,
-    reference_genome: ReferenceGenome,
-    use_gnomad_in_ld_prune: bool,
+    gnomad_qc_ht: hl.Table | None,
 ) -> hl.MatrixTable:
     mt = filter_rows_for_qc(
         mt,
         min_af=0.001,
         min_callrate=0.99,
         apply_hard_filters=False,
     )
-    if not use_gnomad_in_ld_prune:
+    if not gnomad_qc_ht:
```

Comment:
I ported the `read_table` into the relatedness task (so the io looks like it does everywhere else) and simplified the logic to "if the table is available use it, else use `ld_prune`).